src/Sparrow/Json/JsonOperationContext.cs

```diff
@@ -311,12 +311,12 @@ public static ManagedPinnedBuffer ShortLivedInstance()
         public static JsonOperationContext ShortTermSingleUse()
         {
             return new JsonOperationContext(4096, 1024, SharedMultipleUseFlag.None);
-        }
+        }       
 
         public JsonOperationContext(int initialSize, int longLivedSize, SharedMultipleUseFlag lowMemoryFlag)
         {
             Debug.Assert(lowMemoryFlag != null);
-            _disposeOnceRunner = new DisposeOnce<ExceptionRetry>(() =>
+            _disposeOnceRunner = new DisposeOnce<SingleAttemptWithWaitForDisposeToFinish>(() =>
```

Comment:
Let's just change the behavior of `SingleAttempt` to allow recursive calls

Reply:
done